### PR TITLE
Add XMPP server to dev-flood

### PR DESCRIPTION
**Pull request by dkess**
I want to install an XMPP server so we can get Biboumi running, which will allow users to access IRC from XMPP clients (could be good for mobile). It's a Slack alternative that's easy for us to set up and maintain.

We can also potentially provide XMPP as a service to our users.

Next PR will be adding this to actual flood. Then I'll work on Biboumi.

### CODE REVIEW

**Comment by dkess** [comment]
this is tested and working btw

**Comment by dkess** [comment]
According to jvperrin in IRC, we can use DNS to request a cert for `ocf.berkeley.edu`.

**Comment by dkess** [comment]
That will be handled in a later PR, note that this one won't actually install Prosody on `flood` since it only modifies the hiera classes for `dev-flood`.

File: modules/ocf_xmpp/manifests/init.pp [hunk 1]
```diff
@@ -0,0 +1,57 @@
+class ocf_xmpp {
+  # Needed to connect to MySQL
+  package { ['lua-dbi-mysql']: }
+
+  user { 'prosody':
```
Comment (fydai) [inline modules/ocf_xmpp/manifests/init.pp]: use ocfprosody to not conflict with usernames?

Comment (dkess) [reply modules/ocf_xmpp/manifests/init.pp]: The `prosody` name has been reserved for a while, and this is the default name used by the package anyways.

File: modules/ocf_xmpp/manifests/init.pp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        chain  => 'PUPPET-INPUT',
+        proto  => 'tcp',
+        dport  => [5222, 5269],
+        action => 'accept',
+      }
+  };
+
+  $vhost_name = $::host_env ? {
+    'dev'  => 'dev-xmpp.ocf.berkeley.edu',
+    'prod' => 'ocf.berkeley.edu',
```
Comment (kpengboy) [inline modules/ocf_xmpp/manifests/init.pp]: I think this should be `xmpp.ocf.berkeley.edu`?

Comment (dkess) [reply modules/ocf_xmpp/manifests/init.pp]: Yeah, this is correct

**Pull request merged**